generator/centers_table_builder.cpp

```diff
@@ -0,0 +1,61 @@
+#include "generator/centers_table_builder.hpp"
+
+#include "search/search_trie.hpp"
+
+#include "indexer/centers_table.hpp"
+#include "indexer/feature_algo.hpp"
+#include "indexer/features_offsets_table.hpp"
+#include "indexer/features_vector.hpp"
+
+#include "coding/file_container.hpp"
+
+#include "platform/mwm_traits.hpp"
+#include "platform/platform.hpp"
+
+#include "defines.hpp"
+
+namespace indexer
+{
+bool BuildCentersTableFromDataFile(string const & filename, bool forceRebuild)
+{
+  search::CentersTableBuilder builder;
+
+  {
+    auto const & platform = GetPlatform();
+    FilesContainerR rcont(platform.GetReader(filename, "f"));
```

Comment:
Vladimir Byko-Ianko <notifications@github.com> writes:

> In generator/centers_table_builder.cpp:
>
>> +#include "coding/file_container.hpp"
>> +
>> +#include "platform/mwm_traits.hpp"
>> +#include "platform/platform.hpp"
>> +
>> +#include "defines.hpp"
>> +
>> +namespace indexer
>> +{
>> +bool BuildCentersTableFromDataFile(string const & filename, bool forceRebuild)
>> +{
>> +  search::CentersTableBuilder builder;
>> +
>> +  {
>> +    auto const & platform = GetPlatform();
>> +    FilesContainerR rcont(platform.GetReader(filename, "f"));
>
> Platform::GetReader(...) calls Platform::ReadPathForFile(...) and Platform::ReadPathForFile could throw MYTHROW(FileAbsentException, ... .
> Could you please write try/catch in BuildCentersTableFromDataFile?

Usage of platform is replaced by direct creation of FilesContainerR from
filename - it's possible. But this just hides the problem, not fixes it.
Note that Readers and Writers may throw exceptions too (no space left on
device, corrupted filesystem, etc), but this code is going to be used on
server, during maps generation, so do we need to catch and properly
handle exceptions here? Just for consistency with the method signature
(returns false in case of errors), I'm going to add an exception handler
here, but the method still won't be noexcept.

>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.*
>

-- 
Best regards, Yuri